### PR TITLE
added CDN blog post

### PR DESCRIPTION
**Pull request by Alez87**
Added a blog post for the demo about Content Delivery Network.

### CODE REVIEW

File: content/blog/2021-03-26-cdn-demo.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  - Region-aware server
+  - DDoS prevention
+
+
+## General CDN vs. Zenoh CDN
+The architecture and the geographical coverage and presence of a CDN may differ in relation to the content and the service to serve but the basic concepts and building blocks are the following:
+- **Origin server.** The server that produces the initial content.
+- **Point of presence (POP).** Each region is served by a CDN based on a POP that can be considered as an edge server that manages the traffic for the region and guarantees the local CDN functions such as caching, security (in particular DDOS prevention), networking and content delivery. 
+- **Users.** Within a certain region, users connect to the corresponding edge server instead of the origin server.
+- When the user request data, based on monitoring and metrics, those data are trasfered from the origin server to the edge server.
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:32]: Typo: When the user request**s** ...

File: content/blog/2021-03-26-cdn-demo.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+Here a schematic architecture of a CDN:
+
+![msg-sec](../../img/cdn.png)
+
+[**Zenoh**](https://github.com/eclipse-zenoh/zenoh) is a multi-functions protocol that provides pub/sub messaging, geo-distributed storages, queries and computations. The flexibility of Zenoh primitives allows to adopt it also in a wide range of usecases, only apparently far away from it, such as a CDN implementation.
+
+The main differences between implementing a CDN with Zenoh or with a typical mainstream approach are resumed as follows.
+
+**General CDN vs. Zenoh-based CDN** 
+- **Flat Organization.** Zenoh is agnostic from the node where it runs and through the Zenoh configuration it is possbile to easily specify the network organization and topology. Thus, the origin server is considered just as a node with Zenoh router running on it, similarly, to every edge server.
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:43]: "...with **a** Zenoh router..."

File: content/blog/2021-03-26-cdn-demo.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+- **Edge server.** It’s a server with a running Zenoh router, connected to the origin server.
+- **Region.** Area where every connection is routed towards the nearest edge server. Ideally, every edge server is responsible for a certain region and every user wiht a certain region connects to the same edge server.
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:45]: Typo: "wiht" => "with"

File: content/blog/2021-03-26-cdn-demo.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+## Initial configuration
+
+Zenoh and the related projects in its ecosystem are open-source projects available on github. For this demo we use the following projects
+
+{{<table "table table-striped table-bordered">}}
+| Projects                   |  Overview                                        |
+|:---------------------------|:-------------------------------------------------|
+| **Zenoh**                  | Zero overhead pub/sub, store/query and compute protocol [[source code]](github.com/eclipse-zenoh/zenoh)  |
+| **Zenoh-backend-fielsystem** | Backend and Storages for zenoh using the file system [[source code]](github.com/eclipse-zenoh/zenoh-backend-filesystem) |
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:99]: Typo: "Zenoh-backend-fielsystem" => "Zenoh-backend-filesystem"

File: content/blog/2021-03-26-cdn-demo.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+| **Zenoh-python**            | Python client API for zenoh [[source code]](github.com/eclipse-zenoh/zenoh-python) |
+{{</table>}}
+
+
+and we initialize each node with the minimal set of tools to be used:
+
+1.	Install Cargo and Rust (nightly-build)
+2.	Build the “Zenoh” project and the “Zenoh-backend-filesystem” project
+    > **Warning:** the 2 project must be built with the same Rustc version to enable compatilities between them.
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:108]: Now all the zenoh projects contain a `rust-toolchain` file specifying the same Rustc version. It will be automatically be used when running `cargo build`.
=> Warning not needed

File: content/blog/2021-03-26-cdn-demo.md [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+
+## Run the Demo
+
+**Execute Zenoh router for each server**
+    
+    # Set the router options:
+
+    $ address="tcp/<ip_node>"
+    $ port="7447"
+    $ api_port="8001"
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:121]: Out of curiosity: why do you change the REST API port to 8001 (default being 8000) ?

File: content/blog/2021-03-26-cdn-demo.md [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+
+In particular, the option -e of Zenoh is used to connect the routers together. This option is required for edge servers that want to connect to the origin server or P2P/mesh configuration and is not needed on the origin server.
+
+
+**Creates the backend on all Zenoh routers and the storage only on the origin server** 
+
+In fact the edge servers are activated to dispatch the content only if local users request it, thus, initially, the edge servers do not have any storages which are created on demand in relation to the traffic on the region.
+[Check how to create backend and storage on Zenoh](https://github.com/eclipse-zenoh/zenoh-backend-filesystem#file-system-backend-for-eclipse-zenoh)
+
+After adding the backend on the origin server and on the edge server, Zenoh routers will consider *$ZBACKEND_FS_ROOT* directory as the root folder where to store future data.
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:136]: You should probably introduce the `$ZBACKEND_FS_ROOT` env variable and advise to set it before starting the router (or explaining the behaviour if not set).

File: content/blog/2021-03-26-cdn-demo.md [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+
+Here an example about the creation of backend and storage through REST api and Python client.
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:138]: "Here and example..." => "Here **is** an example..."

File: content/blog/2021-03-26-cdn-demo.md [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+    > **Warning:** the 2 project must be built with the same Rustc version to enable compatilities between them.
+3.	Compile both projects and setup Zenoh with the zbackend_fs library in ~/.zenoh/lib ([see the doc]( https://github.com/eclipse-zenoh/zenoh-backend-filesystem/blob/master/README.md#file-system-backend-for-eclipse-zenoh))
+4.	Install Python3 for the client calls and setup the Zenoh-python project ([see the doc](https://github.com/eclipse-zenoh/zenoh-python/blob/master/README.md#eclipse-zenoh-python-api))
+
+
+## Run the Demo
+
+**Execute Zenoh router for each server**
+    
+    # Set the router options:
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:117]: Surround the commands with Markdown code block tags:
\```bash 
    ... 
\```

File: content/blog/2021-03-26-cdn-demo.md [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+**Creates the backend on all Zenoh routers and the storage only on the origin server** 
+
+In fact the edge servers are activated to dispatch the content only if local users request it, thus, initially, the edge servers do not have any storages which are created on demand in relation to the traffic on the region.
+[Check how to create backend and storage on Zenoh](https://github.com/eclipse-zenoh/zenoh-backend-filesystem#file-system-backend-for-eclipse-zenoh)
+
+After adding the backend on the origin server and on the edge server, Zenoh routers will consider *$ZBACKEND_FS_ROOT* directory as the root folder where to store future data.
+
+Here an example about the creation of backend and storage through REST api and Python client.
+
+    # Environment variables:
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:140]: Surround the commands with Markdown code block tags:
\```bash 
    ... 
\```

File: content/blog/2021-03-26-cdn-demo.md [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+    $ path_expr="/demo/example/**"		//storage path expression
+    $ path_prefix="/demo/example"	 	//prefix to strip form the path_expre
+    $ storage_folder="fs_storage"           //directory to store the file
+    $ id="local"                           //identifier for the backend/storage
+
+    # Backend - REST api:
+    $ curl -X PUT -H 'content-type:application/properties' http://$address:8001/@/router/
+    local/plugin/storages/backend/fs
+    
+    # Backend - Python client:
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:151]: Surround the Python code with Markdown code block tags:
\```python 
    ... 
\```

File: content/blog/2021-03-26-cdn-demo.md [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+
+
+**Client writes some data on Origin server**
+
+When a Client creates a content, the Zenoh router is called to store it through the PUT api, by using a key-value pair:
+
+    $ locator="/demo/example/content"
+    $ value="Some_data" //string, int, image or custom file
+
+    # Python code fo the PUT api
```
Comment (JEnoch) [inline content/blog/2021-03-26-cdn-demo.md:183]: Surround the code with Markdown code block tags:
\```python 
    ... 
\```

File: content/blog/2021-03-26-cdn-demo.md [hunk 13]
```diff
@@ -0,0 +1,254 @@
...
+title: "Content Delivery Network with Zenoh"
+date: 2021-03-27
+menu: "blog"
+weight: 300
+description: "27 March 2021 -- Paris."
+draft: false
+---
+
+## Why a Content Delivery Network?
+A Content Delivery Network (CDN) is a popular solution based on geographically distributed servers that work together by collecting the data to the user’s nearest node. Edge servers, located close to the end users, are the base infrastructure a CDN is built upon.
```
Comment (kydos) [inline content/blog/2021-03-26-cdn-demo.md:11]: Instead of "by collecting the data..." I'd say: " by storing the data " or perhaps "by caching the data"

File: content/blog/2021-03-26-cdn-demo.md [hunk 14]
```diff
@@ -0,0 +1,254 @@
...
+
+
+## General CDN vs. Zenoh CDN
+The architecture and the geographical coverage and presence of a CDN may differ in relation to the content and the service to serve but the basic concepts and building blocks are the following:
+- **Origin server.** The server that produces the initial content.
+- **Point of presence (POP).** Each region is served by a CDN based on a POP that can be considered as an edge server that manages the traffic for the region and guarantees the local CDN functions such as caching, security (in particular DDOS prevention), networking and content delivery. 
+- **Users.** Within a certain region, users connect to the corresponding edge server instead of the origin server.
+- When the user request data, based on monitoring and metrics, those data are trasfered from the origin server to the edge server.
+
+Here a schematic architecture of a CDN:
```
Comment (kydos) [inline content/blog/2021-03-26-cdn-demo.md:34]: Perhaps:

"Below is depicted the schematic architecture of a CDN:"

File: content/blog/2021-03-26-cdn-demo.md [hunk 15]
```diff
@@ -0,0 +1,254 @@
...
+
+![msg-sec](../../img/cdn.png)
+
+[**Zenoh**](https://github.com/eclipse-zenoh/zenoh) is a multi-functions protocol that provides pub/sub messaging, geo-distributed storages, queries and computations. The flexibility of Zenoh primitives allows to adopt it also in a wide range of usecases, only apparently far away from it, such as a CDN implementation.
```
Comment (kydos) [inline content/blog/2021-03-26-cdn-demo.md:38]: I would rephrase as follows:

[**Zenoh**](https://github.com/eclipse-zenoh/zenoh) is a protocol that unifies data in motion, data at rest and computations. It  blends traditional pub/sub with geo-distributed storages, queries and computations, while retaining a level of time and space efficiency that is well beyond any of the mainstream stacks.  Zenoh primitives allow cover  a wide range of use-cases, and as shown in this post, make a CDN implementation  rather straightforward.

File: content/blog/2021-03-26-cdn-demo.md (hunk 2 above)
Comment (kydos) [inline content/blog/2021-03-26-cdn-demo.md:43]: I would rephrase this as follows:

 **Location Transparency.** Zenoh provides location transparency and shields the user for the topological details of the network as well as from the specific location of data.  Additionally, in resolving a query Zenoh always favours the set of storages that are closer to the source of the query.

File: content/blog/2021-03-26-cdn-demo.md [hunk 16]
```diff
@@ -0,0 +1,254 @@
...
+
+The main differences between implementing a CDN with Zenoh or with a typical mainstream approach are resumed as follows.
+
+**General CDN vs. Zenoh-based CDN** 
+- **Flat Organization.** Zenoh is agnostic from the node where it runs and through the Zenoh configuration it is possbile to easily specify the network organization and topology. Thus, the origin server is considered just as a node with Zenoh router running on it, similarly, to every edge server.
+- **Edge server.** It’s a server with a running Zenoh router, connected to the origin server.
```
Comment (kydos) [inline content/blog/2021-03-26-cdn-demo.md:44]: The "connected" seem to infer a direct connection. While this is not required.

File: content/blog/2021-03-26-cdn-demo.md [hunk 17]
```diff
@@ -0,0 +1,254 @@
...
+- **Users.** Any device that wants to connect (physical sensors, IoT devices, mobile devices, etc.)
+
+**Benefits of using Zenoh**
+- **Architecture.** Support for both hierarchical and flat organization, just by handling the configuration specified for Zenoh.
+- **Synchronization.** Zenoh natively provides the required features to synchronize contents although is also a pub/sub protocol.
+
+
+## Users involved
+
+Before talking of how to build a CDN, it’s crucial to understand the roles for the different users involved. Users may generally be divided into three levels of domain:
```
Comment (kydos) [inline content/blog/2021-03-26-cdn-demo.md:55]: Before explaining how to build a CDN, ...
